model/Check/Instance/WriteConfigDataCheck.php

```diff
@@ -47,16 +46,22 @@ protected function doCheck(): Report
         foreach ($dir as $fileinfo) {
             if ($fileinfo->isDir() && !$fileinfo->isDot()) {
                 if (!is_writable($fileinfo->getPathname())) {
-                    return new Report(Report::TYPE_ERROR, __('TAO has no permissions to write into \'config/%s\' folder', $fileinfo->getFilename()));
+                    $message = __('TAO has no permissions to write into \'config/%s\' folder', $fileinfo->getFilename());
+                    $this->logError($message);
+                    return new Report(Report::TYPE_ERROR, $message);
                 }
             }
         }
 
         if (!file_exists($dataDir)) {
-            return new Report(Report::TYPE_ERROR, __('\'data\' folder does not exist'));
+            $message = __('\'data\' folder does not exist');
+            $this->logError($message);
+            return new Report(Report::TYPE_ERROR, $message);
         }
         if (!is_writable($dataDir)) {
-            return new Report(Report::TYPE_ERROR, __('TAO has no permissions to write into \'data\' folder'));
+            $message = __('TAO has no permissions to write into \'data\' folder');
+            $this->logError($message);
+            return new Report(Report::TYPE_ERROR, $message);
```

Comment:
I would also recommend to run `is_dir($dataDir)` check. Since on *nix "everything is a file" -> existing file with name `data` would produce "a lot of fun"